mitxgraders/helpers/mitmath/mathfuncs.py

```diff
@@ -236,16 +249,29 @@ def has_one_scalar_input(display_name):
 ARRAY_FUNCTIONS = {
     're': real,
     'im': imag,
-    'conj': np.conj,
-    'norm': np.linalg.norm
+    'conj': np.conj
+}
+
+def has_one_square_input(display_name):
+    return SpecifyDomain.make_decorator('square', display_name=display_name)
+
+ARRAY_ONLY_FUNCTIONS = {
+    'norm': np.linalg.norm,
+    'abs': np.linalg.norm,
```

Comment:
Can we split out `abs` into a function that only works on scalars and vectors (column/row)? Taking the absolute value of a matrix should return an error.